File: request.js
var Request = function (string) {
  var params = Request.regexp.exec(string)
  if (params) {
    this.method = params[1] || 'get'
    this.time = params[2]
    this.path = params[3]
    if (params[4])
      this.body = JSON.parse(params[4])
  } else
    throw 'invalid request'
}

Request.regexp = /(?:(\w+)\:)?(?:(\d+)@)?(\/(?:[^?]|\\\?)*)(?:\?(.*))?/

module.exports = Request

File: request.rb
class Request
  REGEXP = /(?:(\w+)\:)?(?:(\d+)@)?(\/(?:[^?]|\\\?)*)(?:\?(.*))?/

  attr_reader :method, :time, :path, :body

  def initialize string
    params = string.scan REGEXP
    if params.empty?
      raise 'invalid request'
    else
      params = params[0]
      @method = params[0] || :get
      @time = params[1]
      @path = params[2]
      if body = params[3]
        @body = JSON.parse body
      end
    end
  end

end
